src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -320,52 +296,29 @@ void addArg(RefKind refKind, BoundExpression expression)
                         throw ExceptionUtilities.UnexpectedValue(evaluation);
                 }
 
-                BoundExpression makeImplicitIndexArgument(int index, BoundDagTemp lengthTemp)
-                {
-                    return index < 0
-                        ? _factory.IntSubtract(_tempAllocator.GetTemp(lengthTemp), _factory.Literal(-index))
-                        : _factory.Literal(index);
-                }
-
-                BoundExpression makeExplicitIndexArgument(int index, bool fromEnd)
+                BoundExpression makeUnloweredIndexArgument(int index)
                 {
-                    return _factory.New(
-                        WellKnownMember.System_Index__ctor,
-                        ImmutableArray.Create<BoundExpression>(
-                            _factory.Literal(Math.Abs(index)),
-                            _factory.Literal(fromEnd)));
-                }
+                    // LocalRewriter.MakePatternIndexOffsetExpression understands this format
+                    if (index < 0)
+                    {
+                        MethodSymbol symbolOpt = _factory.WellKnownMember(WellKnownMember.System_Index__ctor) as MethodSymbol;
+                        return new BoundFromEndIndexExpression(_factory.Syntax, _factory.Literal(-index),
+                            methodOpt: symbolOpt, _factory.WellKnownType(WellKnownType.System_Index));
+                    }
 
-                BoundExpression makeExplicitRangeArgument(BoundDagSliceEvaluation e)
-                {
-                    return _factory.New(
-                        WellKnownMember.System_Range__ctor,
-                        ImmutableArray.Create(
-                            makeExplicitIndexArgument(e.StartIndex, fromEnd: false),
-                            makeExplicitIndexArgument(e.EndIndex, fromEnd: true)));
+                    return _factory.Convert(_factory.WellKnownType(WellKnownType.System_Index), _factory.Literal(index));
```

Comment:
>_factory.Convert(_factory.WellKnownType(WellKnownType.System_Index), _factory.Literal(index)); [](http://example.com/codeflow?start=27&length=94)

Is doing this going to cause problems when the user-defined conversion operator is missing?